pkg/skaffold/runner/v1/deploy.go

```diff
@@ -45,11 +45,11 @@ func (r *SkaffoldRunner) DeployAndLog(ctx context.Context, out io.Writer, artifa
 	// Update which images are logged.
 	r.AddTagsToPodSelector(artifacts)
 
-	logger := r.createLogger(out, artifacts)
-	defer logger.Stop()
+	defer r.deployer.StopLogger()
 
 	// Logs should be retrieved up to just before the deploy
-	logger.SetSince(time.Now())
+	r.deployer.SetSince(time.Now())
+	r.deployer.RegisterBuildArtifacts(artifacts)
```

Comment:
Method name `RegisterBuildArtifacts` loses the context of it being part of the `Logger` interface after being embedded directly in the `Deployer` interface. Maybe call this method `AddToLogger(artifacts []graph.Artifact)` instead?